### DISCUSSION TITLE
License of the source code

### DISCUSSION THREAD

**Discussion by Kangmo** [magiblot/tvision#170] [Q&A]
Hi magiblot, 

Thanks for the great project, I like it so much and use it on my own server. 

Before using the source code of your work, I would like to check the license of the source code you started to working on this project. 

Based on the following link, Borland released the original code as public domain, but some ports changed license to GPL, and some BSD like license. 

So, could you let me know when you started this project which source code you used? I am going to start a new project using your port and hoping it was either from Borland(public domain) or a port with BSD like license. 

Thanks,
Kangmo
--

https://tvision.sourceforge.net/#wcp

What about copyrights?
The original code is copyrighted by Borland but is freely available from the net. Try [here](ftp://ftp.borland.com/pub/borlandcpp/devsupport/archive/turbovision/tv.zip).
This port is distributed under the GPL license and the Sigala's port under a BSD like license.
According to a FAQ entry in the Borland's site (was in http://www.inprise.com/devsupport/bcppbuilder/faq/QNA906.html when I saw it) the code is public domain. I also asked in the Borland's newsgroup and the TeamB people (not official people but they are who give technical support in the net) said me the FAQ was right.

**Comment by magiblot** [comment]
Hi @Kangmo!

I'm glad that you find this project useful.

The initial code used in this project was Borland's "public domain" release of Turbo Vision's source code. All changes made on top of that are licensed under the MIT license, as I tried to express in the [COPYRIGHT](https://github.com/magiblot/tvision/blob/50ef535b3ca6a7b0634cdf008ff3885126dc3cd3/COPYRIGHT) file.

The explanations about GPL licensing in the link you provided do not apply to this project.

**Reply by Kangmo** [reply]
magiblot, that's great news. I am going to develop a simple IDE using your project, I hope to contribute bug fixes of the tvision project. Again, thanks for the great work. BTW, is this repo having all the changes starting from the Borland's "public domain" release of Turbo Vision's source code? I mean does this repo have the borland's initial source code committed so that all the changes from it can be seen in the repo?

**Reply by magiblot** [reply]
Yes, the first two commits in this repository are Turbo Vision's original source code.

**Comment by Kangmo** [comment]
Thank you for the confirmation, magiblot. It is also great to see code
changes from turbo vision v1 to v2.
